### PR TITLE
listener: respect address.pipe.mode (it didn't work)

### PR DESCRIPTION
**Pull request by sorah**
`listener[].address.pipe.mode` configuration wasn't working since it has been landed, as it was ignored during resolution from config object to Address instance.

----

This patch fixes the following configuration working as expected. 

```yaml
static_resources:
  listeners:
    - name: http
      address:
        pipe:
          path: '/tmp/envoy.sock'
          mode: 438 # 0666
      # ...
```

Commit Message: listener: respect address.pipe.mode (it didn't work)
Risk Level: Low
Testing: added integration test
Docs Changes: N/A
Release Notes: N/A
Fixes: https://github.com/envoyproxy/envoy/issues/11809

### CODE REVIEW

**Comment by sorah** [comment]
I thought that some commit broke this feature, but I couldn't confirm the `mode` configuration was working as expected at the commit https://github.com/envoyproxy/envoy/pull/8423 ...

**Comment by sorah** [comment]
Trying to add an integration test as it has never worked properly.

(PR https://github.com/envoyproxy/envoy/pull/8423 is changing `protobufAddressToAddress`, but listeners were created using `resolveProtoAddress` since when it has been merged)

File: test/integration/uds_integration_test.cc [hunk 1]
```diff
@@ -84,6 +93,24 @@ HttpIntegrationTest::ConnectionCreationFunction UdsListenerIntegrationTest::crea
...
 
+TEST_P(UdsListenerIntegrationTest, TestSocketMode) {
+  if (abstract_namespace_) {
+    // stat(2) against sockets in abstract namespace is not possible
+    GTEST_SKIP();
+  }
+
+  initialize();
+  struct stat listener_stat;
+  EXPECT_EQ(::stat(getListenerSocketName().c_str(), &listener_stat), 0);
```
Comment (lizan) [inline test/integration/uds_integration_test.cc:104]: use `OsSysCallsSingleton` for `stat` syscall?

File: test/integration/uds_integration_test.cc [hunk 2]
```diff
@@ -47,14 +49,20 @@ TEST_P(UdsUpstreamIntegrationTest, RouterDownstreamDisconnectBeforeResponseCompl
...
     TestParameters, UdsListenerIntegrationTest,
     testing::Combine(testing::ValuesIn(TestEnvironment::getIpVersionsForTest()),
-                     testing::Values(false, true)));
+                     testing::Values(false, true), testing::Values(0)));
 #else
 INSTANTIATE_TEST_SUITE_P(
     TestParameters, UdsListenerIntegrationTest,
     testing::Combine(testing::ValuesIn(TestEnvironment::getIpVersionsForTest()),
-                     testing::Values(false)));
+                     testing::Values(false, true), testing::Values(0)));
```
[the remark this answers is not in the document]
Comment (sorah) [reply test/integration/uds_integration_test.cc]: oops, forgot to remove when pasting the change in the above line https://github.com/envoyproxy/envoy/pull/13493/commits/ccd4f0254a9027fec42aae80a12fd441388da91d

File: test/integration/uds_integration_test.cc (hunk 1 above)
Comment (sorah) [reply test/integration/uds_integration_test.cc]: overlooked that, fixed https://github.com/envoyproxy/envoy/pull/13493/commits/16aa22fbdef81a8a25ae1e724853a9537c3d6427

**Comment by sorah** [comment]
CIs passed.

**Review by lizan — APPROVED** [review]
Can you merge main branch once?

**Pull request merged**